File: COLLEGE/ASSIGNMENT8/python/Q.10.py
def find_triplet(arr, target):
    arr.sort()
    for i in range(len(arr) - 2):
        left, right = i + 1, len(arr) - 1
        while left < right:
            current_sum = arr[i] + arr[left] + arr[right]
            if current_sum == target:
                return [arr[i], arr[left], arr[right]]
            elif current_sum < target:
                left += 1
            else:
                right -= 1
    return []

# Test
arr = [1, 4, 45, 6, 10, 8]
target = 22
triplet = find_triplet(arr, target)
print(f"Triplet with sum {target}: {triplet}")

File: COLLEGE/ASSIGNMENT8/c/Q.10.c
#include <stdio.h>

void findTriplet(int arr[], int n, int K) {
    for (int i = 0; i < n - 2; i++) {
        for (int j = i + 1; j < n - 1; j++) {
            for (int k = j + 1; k < n; k++) {
                if (arr[i] + arr[j] + arr[k] == K) {
                    printf("Triplet: %d, %d, %d\n", arr[i], arr[j], arr[k]);
                    return;
                }
            }
        }
    }
    printf("No triplet found\n");
}

int main() {
    int arr[] = {1, 4, 45, 6, 10, 8};
    int K = 22;
    int n = sizeof(arr) / sizeof(arr[0]);

    findTriplet(arr, n, K);
    return 0;
}
